include/EVT/dev/encoder.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef _EVT_ENCODER_
+#define _EVT_ENCODER_
+
+#include <EVT/io/GPIO.hpp>
+#include <EVT/utils/log.hpp>
+#include <EVT/utils/time.hpp>
+
+namespace time = EVT::core::time;
+
+namespace EVT::core::DEV {
+
+/**
+ * This class represents a CTS Series 288 Rotary encoder with a 2-bit Binary resolution.
+ * The static wrappers for the interrupt handles must be created in the main method
+ * see samples/encoder/main.cpp for example implementation.
+ * update() must be called within the main loop to update encoder position.
+ */
+class Encoder {
+public:
+    /**
+     * Constructs an Encoder instance.
+     *
+     * @param[in] a a pin of the encoder
+     * @param[in] b b pin of the encoder
+     * @param range range of the encoder positions
+     * @param initialPosition initial position the encoder is in
+     */
+    Encoder(IO::GPIO& a, IO::GPIO& b, uint64_t range, uint64_t initialPosition, bool rollOver);
+
+    /**
+     * Reads and updates the encoder position.
+     */
+    void update();
+
+    /**
+     * Returns the current absolute position
+     *
+     * @return the current position of the encoder, between 0 and range, inclusive
+     */
+    uint64_t getPosition() const;
+
+    /**
+     * Handles the interrupt generated by pin a. If two interrupts occur too soon after one another the second is ignored.
+     *
+     * @param pin pin that cause the interrupt
+     */
+    void aInterruptHandler(IO::GPIO* pin);
```

Comment:
You don't need to pass the GPIO in here. You don't use it

Reply:
It's necessary because registerIRQ requires you to to pass in a void method that takes in a GPIO pointer. If it does not have a GPIO pointer it does not work.